### PR TITLE
Laboratory work #4, Lubov Gorbunova - 21FPL2

### PR DESCRIPTION
**Pull request by lubagorbunova**

### CODE REVIEW

File: lab_3_keywords_textrank/main.py [hunk 1]
```diff
@@ -258,6 +258,23 @@ def __init__(self) -> None:
         self._position_weights = {}
 
     # Step 4.2
+    def add_edge_help(self, vertex: int, other_vertex: int) -> None:
```
Comment (artyomtugaryov) [inline lab_3_keywords_textrank/main.py:261]: Why do you add the method to the class from LAB3?

File: lab_4_summarization_textrank/main.py [hunk 2]
```diff
@@ -84,31 +102,56 @@ def __init__(self, stop_words: tuple[str, ...], punctuation: tuple[str, ...]) ->
...
         """
         Splits the provided text by sentence
         :param text: the raw text
         :return: a sequence of sentences
         """
-        pass
+        if not isinstance(text, str):
+            raise ValueError
+        text = text.replace('\n', ' ').strip()
+        sentences = re.split(r"(?<=[.!?])\s+(?=[A-ZА-Я])", text)
```
Comment (artyomtugaryov) [inline lab_4_summarization_textrank/main.py:125]: Be ready to explain this expression on the defends

File: lab_4_summarization_textrank/main.py [hunk 3]
```diff
@@ -84,31 +102,56 @@ def __init__(self, stop_words: tuple[str, ...], punctuation: tuple[str, ...]) ->
...
+        sentence_list = []
+        count = 0
+        for item in sentences:
```
Comment (artyomtugaryov) [inline lab_4_summarization_textrank/main.py:128]: what about enumerate?

**Review by artyomtugaryov — CHANGES_REQUESTED** [review]

File: lab_3_keywords_textrank/main.py (hunk 1 above)
Comment (lubagorbunova) [reply lab_3_keywords_textrank/main.py:261]: I'm not sure about adding a new method. I suppose, I added a space string to resolve style problem, which pycharm has told me about, and then I deleted that string. Then, probably, I commited a wrong file. However, it doesn't affect laboratory work #4 as I do not have to import class AdjacencyMatrixGraph, does it?

File: lab_4_summarization_textrank/main.py (hunk 2 above)
Comment (lubagorbunova) [reply lab_4_summarization_textrank/main.py:125]: from python documentation:
(?<=...) Matches if the current position in the string is preceded by a match for ... that ends at the current position. This is called a positive lookbehind assertion.
[.!?] - list of characters
\s Matches Unicode whitespace characters
(?=...) Matches if ... matches next, but doesn’t consume any of the string. 
This is called a lookahead assertion.
[A-ZА-Я] - symbols from A to Z and from cyrillic А to Я.
Overall, this regular expression describes a fragment of a string, that starts with either ., ! or ?, followed by space and a capital leter.

File: lab_3_keywords_textrank/main.py (hunk 1 above)
Comment (demid5111) [reply lab_3_keywords_textrank/main.py:261]: remove

File: lab_4_summarization_textrank/main.py [hunk 4]
```diff
@@ -20,59 +19,78 @@ def __init__(self, text: str, position: int) -> None:
...
+        return self._preprocessed
 
     def set_encoded(self, encoded_sentence: EncodedSentence) -> None:
         """
         Sets the attribute
         :param encoded_sentence: the encoded sentence (a sequence of numbers)
         :return: None
         """
-        pass
+        if not isinstance(encoded_sentence, tuple) or isinstance(encoded_sentence, bool):
```
Comment (demid5111) [inline lab_4_summarization_textrank/main.py:81]: bool is not tuple (!)

File: lab_4_summarization_textrank/main.py [hunk 5]
```diff
@@ -20,59 +19,78 @@ def __init__(self, text: str, position: int) -> None:
...
+            raise ValueError
+        for item in encoded_sentence:
+            if not isinstance(item, int) or isinstance(item, bool):
```
Comment (demid5111) [inline lab_4_summarization_textrank/main.py:84]: .

File: lab_4_summarization_textrank/main.py (hunk 3 above)
Comment (demid5111) [reply lab_4_summarization_textrank/main.py:128]: for idx, el in enumerate(el)

**Review by demid5111 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**